libs/ui/src/hooks/smartcard_auth/use_dipped_smartcard_auth.ts

```diff
@@ -256,3 +241,111 @@ export function useDippedSmartcardAuth({
       throwIllegalValue(auth, 'status');
   }
 }
+
+async function logAuthEvents(
+  logger: Logger,
+  previousAuth: DippedSmartcardAuth.Auth = {
+    status: 'logged_out',
+    reason: 'machine_locked',
+    bootstrapAuthenticatedAdminSession: /* istanbul ignore next */ () =>
+      undefined,
+  },
+  auth: DippedSmartcardAuth.Auth
+) {
+  switch (previousAuth.status) {
+    case 'logged_out': {
+      if (auth.status === 'logged_in') {
```

Comment:
With dipped smartcard auth, is the transition from `logged_out` straight to `logged_in` possible? Or will you always go from `remove_card` to `logged_in` for successful login? Just wondering whether this case is really necessary

Reply:
Good question! Had me stumped for a second there. 100% code coverage guarantees that this case must be used for something...

I tried commenting it out and it looks like the test for bootstrapping admin auth fails, so I think that's the only case (which means once you remove bootstrapping, this case can indeed be removed)

Reply:
Ah yes, that makes sense! Thanks for looking into that!